.github/workflows/nightly_run.yml

```diff
@@ -0,0 +1,32 @@
+name: Nightly Run
+
+on:
+  schedule:
+    # Run all tests at midnight.
+    - cron: "0 0 * * *"
```

Comment:
I am not sure how exact cron runs for github actions but I'd suggest running those at a more random time like 3:42. midnight seems to be a time that many people might choose, so let's help github by spreading the load?

Reply:
I've changed it to run at 2:30am. Spreading the load is noble, but I expect Github (And Azure) scales up in preparation for this or otherwise mitigates it, so I'd rather not run it at a completely arbitrary time to be honest. But 2:30am seems good?